lib/fluent/event_router.rb

```diff
@@ -182,7 +184,7 @@ def filter_stream(tag, es)
 
         def optimized_filter_stream(tag, es)
           new_es = MultiEventStream.new
-          es.each do |time, record|
+          es.each(unpacker: @unpacker) do |time, record|
```

Comment:
Is this safe? event_router can be called from multiple inputs.

Reply:
good catch! 
https://github.com/fluent/fluentd/pull/2559/commits/b8021e9857a7815ea92faaa0acef11dbfe0a240d